src/stream/tag.rs

```diff
@@ -58,44 +76,66 @@ impl Header {
     fn decode(mut reader: impl io::Read) -> crate::Result<Header> {
         let mut header = [0; 10];
         let nread = reader.read(&mut header)?;
-        if nread < header.len() || &header[0..3] != b"ID3" {
```

Comment:
Most of this code moved into `decode_base_header` to avoid duplicating across `decode` and `async_decode`.  The ext header stuff really makes a mess unfortunately.